src/main/java/com/sayurbox/kale/abtest/ABTestClient.java

```diff
@@ -2,11 +2,14 @@
 
 import com.sayurbox.kale.abtest.client.GetUniverseAllocationResponse;
 
+import java.util.HashMap;
 import java.util.List;
 
 public interface ABTestClient {
 
     List<GetUniverseAllocationResponse> getAllUniverseAllocations(String userId);
     GetUniverseAllocationResponse getUniverseAllocation(String userId, String universeId);
 
+    List<GetUniverseAllocationResponse> getAllUniverseAllocations(String userId, HashMap<String, String> properties);
+    GetUniverseAllocationResponse getUniverseAllocation(String userId, String universeId, HashMap<String, String> properties);
```

Comment:
Should we define enum for the properties, since it's based on contract?

Reply:
Too exposed business context in public :p